Marlin/src/core/temp_types.h

```diff
@@ -70,5 +70,23 @@ struct celsius_t : public fixp<true, 10, 6> {
   explicit celsius_t(kelvin_t k) : fixp((float)k - 273.15f) {}
   explicit celsius_t(fahrenheit_t f) : fixp(((float)f - 32) * 0.5555555556f) {}
 
-  inline celsius_t& operator = (const celsius_t & other) { data = other.data; return *this; }
+  celsius_t& operator=(const celsius_t &other) { data = other.data; return *this; }
+  celsius_t& operator=(const int &val)         { data = val;        return *this; }
```

Comment:
This is wrong. It's a fixed point, you must shift the value here. I don't think it's the right fix for compiler's error report.

Reply:
I expect that I made some errors. Please follow up with any needed repairs. In this case, I assumed that your FP type automatically does the shift for an ordinary `int`. But it sounds like it is preferable to do a direct assign from an `int`…?

Reply:
I don't think you need those operators. That's the job of the parent `fixp` class and it's doing it well (see the attached tests). Replicating the operators here leads to confusion and might changes the expected internal value.
You can direct assign from an int or a float and the `fixp::operator =` will be called in consequence (so it's transparent from the calling site).

I'll fix this on Tuesday, don't worry.

Reply:
I added these operators to get past some compile errors. I never did get past all the errors though….